src/ui/components/SettingsPage.tsx

```diff
@@ -0,0 +1,118 @@
+          currentCollection !== PREVIEW_COLLECTION
+        ) {
+          setCustomCollection(currentCollection);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, []);
+
+  const handleCollectionChange = useCallback(
+    async (event: ChangeEvent<HTMLInputElement>) => {
+      const newCollectionId = event.target.value;
+      setCollectionId(newCollectionId);
+      if (newCollectionId !== "custom") {
+        try {
+          await browser.experiments.nimbus.setCollection(newCollectionId);
+        } catch (error) {
+          console.error(error);
+        }
+      } else {
+        try {
+          await browser.experiments.nimbus.setCollection(customCollection);
+        } catch (error) {
+          console.error(error);
+        }
+      }
+    },
+    [customCollection, collectionId, setCollectionId],
+  );
+
+  const handleCustomChange = useCallback(
+    async (event: ChangeEvent<HTMLInputElement>) => {
+      const newCustomCollection = event.target.value;
+      setCustomCollection(newCustomCollection);
+      if (collectionId === "custom") {
+        try {
+          await browser.experiments.nimbus.setCollection(newCustomCollection);
+        } catch (error) {
+          console.error(error);
+        }
+      }
+    },
+    [collectionId, customCollection, setCustomCollection],
+  );
+
+  return (
+    <div className="main-content">
+      <div className="settings">
+        <p className="settings__title">Settings</p>
+        <hr className="settings__underline" />
+        <p className="settings__radio-group-title">
+          Choose experiment collection
+        </p>
+        <div className="settings__radio-group">
+          <label className="settings__radio-label">
+            <input
+              className="settings__radio"
+              type="radio"
+              value={LIVE_COLLECTION}
+              checked={collectionId === LIVE_COLLECTION}
+              onChange={handleCollectionChange}
+            />
+            Live ({LIVE_COLLECTION})
+          </label>
+          <label className="settings__radio-label">
+            <input
+              className="settings__radio"
+              type="radio"
+              value={PREVIEW_COLLECTION}
+              checked={collectionId === PREVIEW_COLLECTION}
+              onChange={handleCollectionChange}
+            />
+            Preview ({PREVIEW_COLLECTION})
+          </label>
+          <label className="settings__radio-label">
+            <input
+              className="settings__radio"
+              type="radio"
+              value="custom"
+              checked={
+                collectionId !== LIVE_COLLECTION &&
+                collectionId !== PREVIEW_COLLECTION
+              }
+              onChange={handleCollectionChange}
+            />
+            Custom
+            <input
+              type="text"
+              value={customCollection}
+              onChange={handleCustomChange}
```

Comment:
We should disable this input using the same logic as the `checked` attribute on the radio above.